### PR TITLE
Different link scenario

### PR DESCRIPTION
**Pull request by tanyaa-tw**
A-1205810604262869 | https://app.asana.com/0/1204322126657135/1205810604262869/f | Refreshing the 'Treatments' display control under different scenarios of 'Add to Drug Chart' link, along with Edit Drug Chart scenario

### CODE REVIEW

File: src/features/DisplayControls/Treatments/components/Treatments.jsx [hunk 1]
```diff
@@ -101,61 +119,73 @@ const Treatments = (props) => {
...
   const modifyTreatmentData = (drugOrders) => {
-    const treatments = drugOrders.visitDrugOrders
-      .filter((drugOrder) => isIPDDrugOrder(drugOrder))
-      .map((drugOrder) => {
+    const treatments = drugOrders.ipdDrugOrders
+      .filter((drugOrderObject) => isIPDDrugOrder(drugOrderObject))
+      .map((drugOrderObject) => {
+        var isEditDisabled;
+        var showEditDrugChartLink;
+        if (drugOrderObject.drugOrderSchedule != null) {
```
Comment (Arjun-Go) [inline src/features/DisplayControls/Treatments/components/Treatments.jsx:128]: Code can be optimised to get rid of nested if - else
```
if (drugOrderObject.drugOrderSchedule) {
  showEditDrugChartLink = true;
  isEditDisabled = drugOrderObject.drugOrderSchedule.medicationAdministrationStarted;
} else {
  showEditDrugChartLink = false;
  isEditDisabled = false;
}
```

**Review by Arjun-Go — CHANGES_REQUESTED** [review]

File: src/features/DrugChartSlider/components/DrugChartSlider.jsx [hunk 2]
```diff
@@ -364,11 +368,52 @@ const DrugChartSlider = (props) => {
...
+        const startTimeValue = enable24HourTimers
+          ? epochTo24HourTimeFormat(drugOrderSchedule.slotStartTime)
+          : epochTo12HourTimeFormat(drugOrderSchedule.slotStartTime);
+        setStartTime(startTimeValue);
+      }
+
+      if (scheduleTimings.firstDaySlotsStartTime) {
+        setFirstDaySlotsMissed(1);
+        scheduleTimings.firstDaySlotsStartTime.forEach((schedule) => {
+          var frequency = enableSchedule?.frequencyPerDay;
```
Comment (ayush34957) [inline src/features/DrugChartSlider/components/DrugChartSlider.jsx:388]: use let/const and please check for other files too

File: src/features/DrugChartSlider/utils/DrugChartSliderUtils.js [hunk 3]
```diff
@@ -102,3 +114,61 @@ export const getUTCTimeEpoch = (time, enable24HourTimers, scheduledDate) => {
...
+
+export const epochTo24HourTimeFormat = (epochSeconds) => {
+  const date = new Date(epochSeconds * 1000);
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+  const formattedTime = `${hours}:${minutes}`;
+  return formattedTime;
+};
+
+export const epochTo12HourTimeFormat = (epochSeconds) => {
```
Comment (ayush34957) [inline src/features/DrugChartSlider/utils/DrugChartSliderUtils.js:126]: These methods are generic methods so move these methods to src/utils/DateTimeUtils.js file

File: src/features/DrugChartSlider/utils/DrugChartSliderUtils.js [hunk 4]
```diff
@@ -102,3 +114,61 @@ export const getUTCTimeEpoch = (time, enable24HourTimers, scheduledDate) => {
...
+  const date = new Date(epochSeconds * 1000);
+  const hours = date.getHours();
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+  const amOrPm = hours >= 12 ? "PM" : "AM";
+  const formattedHours = (hours % 12 || 12).toString().padStart(2, "0");
+  const formattedTime = `${formattedHours}:${minutes} ${amOrPm}`;
```
Comment (ayush34957) [inline src/features/DrugChartSlider/utils/DrugChartSliderUtils.js:132]: for formatting the dates use moment js methods, moments js has methods to convert epoch to desired date format, no need of writing these logics on our end

**Pull request merged**